internal/features/features.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package features sets the feature gates that
+// helm-controller supports, and their default states.
+package features
+
+import feathelper "github.com/fluxcd/pkg/runtime/features"
+
+const (
+	// When enabled, CacheSecretsAndConfigMaps will allow the controller-runtime
+	// client to exhibit its default behavior of caching all resources that helm-controller
+	// requests. Since this may cause high memory usage and require cluster-wide list
+	// and watch permissions, it is disabled by default.
```

Comment:
```suggestion
	// CacheSecretsAndConfigMaps configures the caching of Secrets and ConfigMaps
	// by the controller-runtime client.
	// 
	// When enabled, it will cache both object types, resulting in increased memory usage
	// and cluster-wide RBAC permissions (list and watch).
```